src/ripple/app/misc/ValidatorList.h

```diff
@@ -399,36 +405,32 @@ ValidatorList::onConsensusStart (
         }
     }
 
-    // This quorum guarantees sufficient overlap with the trusted sets of other
-    // nodes using the same set of published lists.
-    std::size_t quorum = keyListings_.size()/2 + 1;
-
-    // Increment the quorum to prevent two unlisted validators using the same
-    // even number of listed validators from forking.
-    if (localPubKey_.size() && ! localKeyListed &&
-            rankedKeys.size () > 1 && keyListings_.size () % 2 != 0)
-        ++quorum;
+    // This minimum quorum guarantees safe overlap with the trusted sets of
+    // other nodes using the same set of published lists.
+    std::size_t quorum = calculateMinimumQuorum (keyListings_.size(),
+        localPubKey_.size() && !localKeyListed);
 
     JLOG (j_.debug()) <<
         rankedKeys.size() << "  of " << keyListings_.size() <<
         " listed validators eligible for inclusion in the trusted set";
 
     auto size = rankedKeys.size();
 
-    // Use all eligible keys if there is less than 10 listed validators or
-    // only one trusted list
-    if (size < 10 || publisherLists_.size() == 1)
+    // Do not require 80% quorum for less than 10 trusted validators
+    if (rankedKeys.size() >= 10)
     {
-        // Try to raise the quorum toward or above 80% of the trusted set
-        std::size_t const targetQuorum = ValidatorList::calculateQuorum (size);
-        if (targetQuorum > quorum)
-            quorum = targetQuorum;
-    }
-    else
-    {
-        // reduce the trusted set size so that the quorum represents
-        // at least 80%
-        size = quorum * 1.25;
+        // Use all eligible keys if there is only one trusted list
+        if (publisherLists_.size() == 1)
+        {
+            // Try to raise the quorum to at least 80% of the trusted set
+            quorum = std::max(quorum, size - size / 5);
+        }
+        else
+        {
+            // Reduce the trusted set size so that the quorum represents
+            // at least 80%
+            size = quorum * 1.25;
```

Comment:
`size` can be greater than `rankedKeys_.size()`
That loop should be fine in that case since it is iterating `rankedKeys`. The result is just that we end up with fewer trusted keys than `size`.